fetching/strat.py

```diff
@@ -74,7 +74,8 @@ def _apply(self, path: pathlib.Path) -> None:
                 skip_line = False
                 continue
             if PyTestStrategy.FUNCTION_PATTERN.fullmatch(line):
-                lines.insert(i, PyTestStrategy.REGISTER)
+                indentation = line.split("def")[0]
```

Comment:
So no switch to an AST based implementation?

Reply:
I ask because this seems very rudimentary, and not stable at all. 
Please consider using `ast.parse`, `ast.visit` and `ast.NodeVisitor`

Reply:
I ask because this seems very rudimentary.
Please consider using `ast.parse`, `ast.visit` and `ast.NodeVisitor`

Reply:
As discussed, we won't use AST for this fix.

Reply:
After rediscussing :D 
We don't need to detect the whitespace in front of the method, we just need to update the list of decorators.

The Python documentation gives a function definition as

```py
FunctionDef(identifier name, arguments args,
                       stmt* body, expr* decorator_list, expr? returns,
                       string? type_comment)
```

Simply appending to `decorator_list` should cover everything.
I doubt we need to cover AsyncFunctionDefs.